op-service/txmgr/txmgr.go

```diff
@@ -196,6 +217,12 @@ type TxCandidate struct {
 	Value *big.Int
 }
 
+type PendingTxWithCancel struct {
+	tx     *types.Transaction
```

Comment:
It might be nice to extend this to an array because the key for this is a nonce & it will often have multiple transactions (different gas prices + sig, same data) per nonce when resubmitting.